File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/views/inlineaudiorecorder/CometChatInlineAudioRecorderStyle.java
package com.cometchat.chatuikit.shared.views.inlineaudiorecorder;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.annotation.ColorInt;
import androidx.annotation.Dimension;
import androidx.annotation.NonNull;
import androidx.annotation.StyleRes;

import com.cometchat.chatuikit.CometChatTheme;
import com.cometchat.chatuikit.shared.models.BaseStyle;

/**
 * Style configuration class for CometChatInlineAudioRecorder.
 * <p>
 * This class provides a fluent Builder pattern for configuring the visual appearance
 * of the inline audio recorder component, including colors, dimensions, and icons.
 * </p>
 * <p>
 * Use {@link #getDefault(Context)} to get a style with theme-appropriate defaults,
 * or use the {@link Builder} to create a custom configuration.
 * </p>
 *
 * @see CometChatInlineAudioRecorder
 */
public class CometChatInlineAudioRecorderStyle extends BaseStyle {
    private static final String TAG = CometChatInlineAudioRecorderStyle.class.getSimpleName();

    // Colors
    private @ColorInt int backgroundColor;
    private @ColorInt int waveformRecordingColor;
    private @ColorInt int waveformPlayingColor;
    private @ColorInt int waveformUnplayedColor;
    private @ColorInt int durationTextColor;
    private @StyleRes int durationTextAppearance;
    private @ColorInt int deleteButtonIconColor;
    private @ColorInt int sendButtonIconColor;
    private @ColorInt int sendButtonBackgroundColor;
    private @ColorInt int playButtonIconColor;
    private @ColorInt int pauseButtonIconColor;
    private @ColorInt int recordButtonIconColor;
    private @ColorInt int recordingIndicatorColor;

    // Dimensions
    private @Dimension float waveformBarWidth;
    private @Dimension float waveformBarSpacing;
    private @Dimension float waveformMinBarHeight;
    private @Dimension float waveformMaxBarHeight;
    private int waveformMaxBarCount;

    // Icons
    private Drawable deleteIcon;
    private Drawable recordIcon;
    private Drawable pauseIcon;
    private Drawable playIcon;
    private Drawable sendIcon;
    private Drawable micIcon;

    /**
     * Private constructor. Use {@link Builder} to create instances.
     */
    private CometChatInlineAudioRecorderStyle() {
    }


    /**
     * Creates a default style configuration using CometChatTheme defaults.
     *
     * @param context The context used to access theme resources.
     * @return A new CometChatInlineAudioRecorderStyle with theme-appropriate defaults.
     */
    public static CometChatInlineAudioRecorderStyle getDefault(@NonNull Context context) {
        return new Builder()
            .setBackgroundColor(CometChatTheme.getBackgroundColor1(context))
            .setWaveformRecordingColor(CometChatTheme.getPrimaryColor(context))
            .setWaveformPlayingColor(CometChatTheme.getPrimaryColor(context))
            .setWaveformUnplayedColor(CometChatTheme.getNeutralColor300(context))
            .setDurationTextColor(CometChatTheme.getTextColorSecondary(context))
            .setDeleteButtonIconColor(CometChatTheme.getIconTintSecondary(context))
            .setSendButtonIconColor(CometChatTheme.getColorWhite(context))
            .setSendButtonBackgroundColor(CometChatTheme.getPrimaryColor(context))
            .setPlayButtonIconColor(CometChatTheme.getIconTintSecondary(context))
            .setPauseButtonIconColor(CometChatTheme.getIconTintSecondary(context))
            .setRecordButtonIconColor(CometChatTheme.getIconTintSecondary(context))
            .setRecordingIndicatorColor(CometChatTheme.getErrorColor(context))
            .setWaveformBarWidth(3f)
            .setWaveformBarSpacing(2f)
            .setWaveformMinBarHeight(4f)
            .setWaveformMaxBarHeight(32f)
            .setWaveformMaxBarCount(35)
            .build();
    }

    // Getters

    /**
     * Returns the background color of the inline audio recorder.
     *
     * @return The background color.
     */
    public @ColorInt int getBackgroundColor() {
        return backgroundColor;
    }

    /**
     * Returns the waveform color used during recording.
     *
     * @return The recording waveform color.
     */
    public @ColorInt int getWaveformRecordingColor() {
        return waveformRecordingColor;
    }

    /**
     * Returns the waveform color for played portions during playback.
     *
     * @return The playing waveform color.
     */
    public @ColorInt int getWaveformPlayingColor() {
        return waveformPlayingColor;
    }

    /**
     * Returns the waveform color for unplayed portions during playback.
     *
     * @return The unplayed waveform color.
     */
    public @ColorInt int getWaveformUnplayedColor() {
        return waveformUnplayedColor;
    }

    /**
     * Returns the duration text color.
     *
     * @return The duration text color.
     */
    public @ColorInt int getDurationTextColor() {
        return durationTextColor;
    }

    /**
     * Returns the duration text appearance style resource.
     *
     * @return The text appearance style resource ID.
     */
    public @StyleRes int getDurationTextAppearance() {
        return durationTextAppearance;
    }

    /**
     * Returns the delete button icon color.
     *
     * @return The delete button icon color.
     */
    public @ColorInt int getDeleteButtonIconColor() {
        return deleteButtonIconColor;
    }

    /**
     * Returns the send button icon color.
     *
     * @return The send button icon color.
     */
    public @ColorInt int getSendButtonIconColor() {
        return sendButtonIconColor;
    }

    /**
     * Returns the send button background color.
     *
     * @return The send button background color.
     */
    public @ColorInt int getSendButtonBackgroundColor() {
        return sendButtonBackgroundColor;
    }

    /**
     * Returns the play button icon color.
     *
     * @return The play button icon color.
     */
    public @ColorInt int getPlayButtonIconColor() {
        return playButtonIconColor;
    }

    /**
     * Returns the pause button icon color.
     *
     * @return The pause button icon color.
     */
    public @ColorInt int getPauseButtonIconColor() {
        return pauseButtonIconColor;
    }

    /**
     * Returns the record button icon color.
     *
     * @return The record button icon color.
     */
    public @ColorInt int getRecordButtonIconColor() {
        return recordButtonIconColor;
    }

    /**
     * Returns the recording indicator color.
     *
     * @return The recording indicator color.
     */
    public @ColorInt int getRecordingIndicatorColor() {
        return recordingIndicatorColor;
    }

    /**
     * Returns the waveform bar width in dp.
     *
     * @return The waveform bar width.
     */
    public @Dimension float getWaveformBarWidth() {
        return waveformBarWidth;
    }

    /**
     * Returns the waveform bar spacing in dp.
     *
     * @return The waveform bar spacing.
     */
    public @Dimension float getWaveformBarSpacing() {
        return waveformBarSpacing;
    }

    /**
     * Returns the minimum waveform bar height in dp.
     *
     * @return The minimum waveform bar height.
     */
    public @Dimension float getWaveformMinBarHeight() {
        return waveformMinBarHeight;
    }

    /**
     * Returns the maximum waveform bar height in dp.
     *
     * @return The maximum waveform bar height.
     */
    public @Dimension float getWaveformMaxBarHeight() {
        return waveformMaxBarHeight;
    }

    /**
     * Returns the maximum number of waveform bars to display.
     *
     * @return The maximum bar count.
     */
    public int getWaveformMaxBarCount() {
        return waveformMaxBarCount;
    }

    /**
     * Returns the delete icon drawable.
     *
     * @return The delete icon drawable, or null if not set.
     */
    public Drawable getDeleteIcon() {
        return deleteIcon;
    }

    /**
     * Returns the record icon drawable.
     *
     * @return The record icon drawable, or null if not set.
     */
    public Drawable getRecordIcon() {
        return recordIcon;
    }

    /**
     * Returns the pause icon drawable.
     *
     * @return The pause icon drawable, or null if not set.
     */
    public Drawable getPauseIcon() {
        return pauseIcon;
    }

    /**
     * Returns the play icon drawable.
     *
     * @return The play icon drawable, or null if not set.
     */
    public Drawable getPlayIcon() {
        return playIcon;
    }

    /**
     * Returns the send icon drawable.
     *
     * @return The send icon drawable, or null if not set.
     */
    public Drawable getSendIcon() {
        return sendIcon;
    }

    /**
     * Returns the mic icon drawable.
     *
     * @return The mic icon drawable, or null if not set.
     */
    public Drawable getMicIcon() {
        return micIcon;
    }


    /**
     * Builder class for creating CometChatInlineAudioRecorderStyle instances.
     * <p>
     * Provides a fluent API for configuring all style properties.
     * </p>
     */
    public static class Builder {
        private final CometChatInlineAudioRecorderStyle style;

        /**
         * Creates a new Builder instance.
         */
        public Builder() {
            style = new CometChatInlineAudioRecorderStyle();
        }

        /**
         * Sets the background color of the inline audio recorder.
         *
         * @param backgroundColor The background color.
         * @return This Builder for chaining.
         */
        public Builder setBackgroundColor(@ColorInt int backgroundColor) {
            style.backgroundColor = backgroundColor;
            return this;
        }

        /**
         * Sets the waveform color used during recording.
         *
         * @param waveformRecordingColor The recording waveform color.
         * @return This Builder for chaining.
         */
        public Builder setWaveformRecordingColor(@ColorInt int waveformRecordingColor) {
            style.waveformRecordingColor = waveformRecordingColor;
            return this;
        }

        /**
         * Sets the waveform color for played portions during playback.
         *
         * @param waveformPlayingColor The playing waveform color.
         * @return This Builder for chaining.
         */
        public Builder setWaveformPlayingColor(@ColorInt int waveformPlayingColor) {
            style.waveformPlayingColor = waveformPlayingColor;
            return this;
        }

        /**
         * Sets the waveform color for unplayed portions during playback.
         *
         * @param waveformUnplayedColor The unplayed waveform color.
         * @return This Builder for chaining.
         */
        public Builder setWaveformUnplayedColor(@ColorInt int waveformUnplayedColor) {
            style.waveformUnplayedColor = waveformUnplayedColor;
            return this;
        }

        /**
         * Sets the duration text color.
         *
         * @param durationTextColor The duration text color.
         * @return This Builder for chaining.
         */
        public Builder setDurationTextColor(@ColorInt int durationTextColor) {
            style.durationTextColor = durationTextColor;
            return this;
        }

        /**
         * Sets the duration text appearance style resource.
         *
         * @param durationTextAppearance The text appearance style resource ID.
         * @return This Builder for chaining.
         */
        public Builder setDurationTextAppearance(@StyleRes int durationTextAppearance) {
            style.durationTextAppearance = durationTextAppearance;
            return this;
        }

        /**
         * Sets the delete button icon color.
         *
         * @param deleteButtonIconColor The delete button icon color.
         * @return This Builder for chaining.
         */
        public Builder setDeleteButtonIconColor(@ColorInt int deleteButtonIconColor) {
            style.deleteButtonIconColor = deleteButtonIconColor;
            return this;
        }

        /**
         * Sets the send button icon color.
         *
         * @param sendButtonIconColor The send button icon color.
         * @return This Builder for chaining.
         */
        public Builder setSendButtonIconColor(@ColorInt int sendButtonIconColor) {
            style.sendButtonIconColor = sendButtonIconColor;
            return this;
        }

        /**
         * Sets the send button background color.
         *
         * @param sendButtonBackgroundColor The send button background color.
         * @return This Builder for chaining.
         */
        public Builder setSendButtonBackgroundColor(@ColorInt int sendButtonBackgroundColor) {
            style.sendButtonBackgroundColor = sendButtonBackgroundColor;
            return this;
        }

        /**
         * Sets the play button icon color.
         *
         * @param playButtonIconColor The play button icon color.
         * @return This Builder for chaining.
         */
        public Builder setPlayButtonIconColor(@ColorInt int playButtonIconColor) {
            style.playButtonIconColor = playButtonIconColor;
            return this;
        }

        /**
         * Sets the pause button icon color.
         *
         * @param pauseButtonIconColor The pause button icon color.
         * @return This Builder for chaining.
         */
        public Builder setPauseButtonIconColor(@ColorInt int pauseButtonIconColor) {
            style.pauseButtonIconColor = pauseButtonIconColor;
            return this;
        }

        /**
         * Sets the record button icon color.
         *
         * @param recordButtonIconColor The record button icon color.
         * @return This Builder for chaining.
         */
        public Builder setRecordButtonIconColor(@ColorInt int recordButtonIconColor) {
            style.recordButtonIconColor = recordButtonIconColor;
            return this;
        }

        /**
         * Sets the recording indicator color.
         *
         * @param recordingIndicatorColor The recording indicator color.
         * @return This Builder for chaining.
         */
        public Builder setRecordingIndicatorColor(@ColorInt int recordingIndicatorColor) {
            style.recordingIndicatorColor = recordingIndicatorColor;
            return this;
        }

        /**
         * Sets the waveform bar width in dp.
         *
         * @param waveformBarWidth The waveform bar width.
         * @return This Builder for chaining.
         */
        public Builder setWaveformBarWidth(@Dimension float waveformBarWidth) {
            style.waveformBarWidth = waveformBarWidth;
            return this;
        }

        /**
         * Sets the waveform bar spacing in dp.
         *
         * @param waveformBarSpacing The waveform bar spacing.
         * @return This Builder for chaining.
         */
        public Builder setWaveformBarSpacing(@Dimension float waveformBarSpacing) {
            style.waveformBarSpacing = waveformBarSpacing;
            return this;
        }

        /**
         * Sets the minimum waveform bar height in dp.
         *
         * @param waveformMinBarHeight The minimum waveform bar height.
         * @return This Builder for chaining.
         */
        public Builder setWaveformMinBarHeight(@Dimension float waveformMinBarHeight) {
            style.waveformMinBarHeight = waveformMinBarHeight;
            return this;
        }

        /**
         * Sets the maximum waveform bar height in dp.
         *
         * @param waveformMaxBarHeight The maximum waveform bar height.
         * @return This Builder for chaining.
         */
        public Builder setWaveformMaxBarHeight(@Dimension float waveformMaxBarHeight) {
            style.waveformMaxBarHeight = waveformMaxBarHeight;
            return this;
        }

        /**
         * Sets the maximum number of waveform bars to display.
         *
         * @param waveformMaxBarCount The maximum bar count.
         * @return This Builder for chaining.
         */
        public Builder setWaveformMaxBarCount(int waveformMaxBarCount) {
            style.waveformMaxBarCount = waveformMaxBarCount;
            return this;
        }

        /**
         * Sets the delete icon drawable.
         *
         * @param deleteIcon The delete icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setDeleteIcon(Drawable deleteIcon) {
            style.deleteIcon = deleteIcon;
            return this;
        }

        /**
         * Sets the record icon drawable.
         *
         * @param recordIcon The record icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setRecordIcon(Drawable recordIcon) {
            style.recordIcon = recordIcon;
            return this;
        }

        /**
         * Sets the pause icon drawable.
         *
         * @param pauseIcon The pause icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setPauseIcon(Drawable pauseIcon) {
            style.pauseIcon = pauseIcon;
            return this;
        }

        /**
         * Sets the play icon drawable.
         *
         * @param playIcon The play icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setPlayIcon(Drawable playIcon) {
            style.playIcon = playIcon;
            return this;
        }

        /**
         * Sets the send icon drawable.
         *
         * @param sendIcon The send icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setSendIcon(Drawable sendIcon) {
            style.sendIcon = sendIcon;
            return this;
        }

        /**
         * Sets the mic icon drawable.
         *
         * @param micIcon The mic icon drawable.
         * @return This Builder for chaining.
         */
        public Builder setMicIcon(Drawable micIcon) {
            style.micIcon = micIcon;
            return this;
        }

        /**
         * Builds and returns the configured CometChatInlineAudioRecorderStyle.
         *
         * @return The configured style instance.
         */
        public CometChatInlineAudioRecorderStyle build() {
            return style;
        }
    }
}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/presentation/shared/inlineaudiorecorder/CometChatInlineAudioRecorderStyle.kt
package com.cometchat.uikit.kotlin.presentation.shared.inlineaudiorecorder

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.annotation.ColorInt
import androidx.annotation.Dimension
import androidx.annotation.StyleRes
import com.cometchat.uikit.kotlin.theme.CometChatTheme

/**
 * Style configuration for CometChatInlineAudioRecorder.
 * Contains all visual styling properties for the inline audio recorder component.
 *
 * This style class follows the CometChat Compose Component Architecture Guide
 * and provides full customization of the inline audio recorder's appearance.
 *
 * Layout structure (single row):
 * [Delete] [Record/Play] [Waveform] [Duration] [Pause/Mic] [Send]
 *
 * **Validates: Requirements 10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7, 10.8, 10.9, 14.4**
 *
 * @param backgroundColor Background color for the recorder container
 * @param cornerRadius Corner radius of the recorder container
 * @param strokeColor Color of the recorder border/stroke
 * @param strokeWidth Width of the recorder border
 *
 * @param barColor Color for inactive waveform bars (grey)
 * @param recordingBarColor Color for waveform bars during recording (purple)
 * @param playingBarColor Color for active waveform bars during playback (purple)
 * @param barWidth Width of each waveform bar
 * @param barSpacing Spacing between waveform bars
 * @param barMinHeight Minimum height of waveform bars
 * @param barMaxHeight Maximum height of waveform bars
 * @param barCornerRadius Corner radius of waveform bars
 * @param barCount Number of waveform bars to display
 *
 * @param durationTextColor Color for the duration/timer text
 * @param durationTextAppearance Text appearance resource for the duration text
 *
 * @param recordButtonIconColor Icon color for the record button (red)
 * @param recordButtonBackgroundColor Background color for the record button
 * @param recordingIndicatorColor Color for the pulsing red dot indicator during recording
 *
 * @param playButtonIconColor Icon color for the play button
 * @param playButtonBackgroundColor Background color for the play button
 *
 * @param pauseButtonIconColor Icon color for the pause button
 * @param pauseButtonBackgroundColor Background color for the pause button
 *
 * @param deleteButtonIconColor Icon color for the delete button
 * @param deleteButtonBackgroundColor Background color for the delete button
 *
 * @param sendButtonIconColor Icon color for the send button (white)
 * @param sendButtonBackgroundColor Background color for the send button (purple circular)
 *
 * @param micButtonIconColor Icon color for the mic/resume button
 * @param micButtonBackgroundColor Background color for the mic button
 *
 * @param playIcon Drawable for the play icon
 * @param pauseIcon Drawable for the pause icon
 * @param deleteIcon Drawable for the delete icon
 * @param sendIcon Drawable for the send icon
 * @param micIcon Drawable for the mic icon
 */
data class CometChatInlineAudioRecorderStyle(
    // Container styling
    @ColorInt val backgroundColor: Int = 0,
    @Dimension val cornerRadius: Int = 0,
    @ColorInt val strokeColor: Int = 0,
    @Dimension val strokeWidth: Int = 0,
    
    // Waveform styling
    @ColorInt val barColor: Int = 0,
    @ColorInt val recordingBarColor: Int = 0,
    @ColorInt val playingBarColor: Int = 0,
    @Dimension val barWidth: Int = 0,
    @Dimension val barSpacing: Int = 0,
    @Dimension val barMinHeight: Int = 0,
    @Dimension val barMaxHeight: Int = 0,
    @Dimension val barCornerRadius: Int = 0,
    val barCount: Int = 45,
    
    // Duration text styling
    @ColorInt val durationTextColor: Int = 0,
    @StyleRes val durationTextAppearance: Int = 0,
    
    // Record button styling
    @ColorInt val recordButtonIconColor: Int = 0,
    @ColorInt val recordButtonBackgroundColor: Int = 0,
    @ColorInt val recordingIndicatorColor: Int = 0,
    
    // Play button styling
    @ColorInt val playButtonIconColor: Int = 0,
    @ColorInt val playButtonBackgroundColor: Int = 0,
    
    // Pause button styling
    @ColorInt val pauseButtonIconColor: Int = 0,
    @ColorInt val pauseButtonBackgroundColor: Int = 0,
    
    // Delete button styling
    @ColorInt val deleteButtonIconColor: Int = 0,
    @ColorInt val deleteButtonBackgroundColor: Int = 0,
    
    // Send button styling
    @ColorInt val sendButtonIconColor: Int = 0,
    @ColorInt val sendButtonBackgroundColor: Int = 0,
    
    // Mic/Resume button styling
    @ColorInt val micButtonIconColor: Int = 0,
    @ColorInt val micButtonBackgroundColor: Int = 0,
    
    // Icon drawables
    val playIcon: Drawable? = null,
    val pauseIcon: Drawable? = null,
    val deleteIcon: Drawable? = null,
    val sendIcon: Drawable? = null,
    val micIcon: Drawable? = null
) {
    /**
     * Builder class for creating CometChatInlineAudioRecorderStyle instances.
     * Provides a fluent API for setting style properties.
     * Container styling matches MessageComposer compose box for visual consistency.
     */
    class Builder(private val context: Context) {
        // Container styling - matches MessageComposer compose box
        @ColorInt private var backgroundColor: Int = CometChatTheme.getBackgroundColor1(context)
        @Dimension private var cornerRadius: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_corner_radius_2
        )
        @ColorInt private var strokeColor: Int = CometChatTheme.getBorderColorLight(context)
        @Dimension private var strokeWidth: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_1dp
        )
        
        // Waveform styling
        @ColorInt private var barColor: Int = CometChatTheme.getNeutralColor300(context)
        @ColorInt private var recordingBarColor: Int = CometChatTheme.getPrimaryColor(context)
        @ColorInt private var playingBarColor: Int = CometChatTheme.getPrimaryColor(context)
        @Dimension private var barWidth: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_2dp
        )
        @Dimension private var barSpacing: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_2dp
        )
        @Dimension private var barMinHeight: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_2dp
        )
        @Dimension private var barMaxHeight: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_16dp
        )
        @Dimension private var barCornerRadius: Int = context.resources.getDimensionPixelSize(
            com.cometchat.uikit.kotlin.R.dimen.cometchat_1dp
        )
        private var barCount: Int = 45
        
        // Duration text styling
        @ColorInt private var durationTextColor: Int = CometChatTheme.getTextColorSecondary(context)
        @StyleRes private var durationTextAppearance: Int = CometChatTheme.getTextAppearanceBodyRegular(context)
        
        // Record button styling
        @ColorInt private var recordButtonIconColor: Int = CometChatTheme.getErrorColor(context)
        @ColorInt private var recordButtonBackgroundColor: Int = android.graphics.Color.TRANSPARENT
        @ColorInt private var recordingIndicatorColor: Int = CometChatTheme.getErrorColor(context)
        
        // Play button styling
        @ColorInt private var playButtonIconColor: Int = CometChatTheme.getIconTintPrimary(context)
        @ColorInt private var playButtonBackgroundColor: Int = android.graphics.Color.TRANSPARENT
        
        // Pause button styling
        @ColorInt private var pauseButtonIconColor: Int = CometChatTheme.getIconTintPrimary(context)
        @ColorInt private var pauseButtonBackgroundColor: Int = android.graphics.Color.TRANSPARENT
        
        // Delete button styling
        @ColorInt private var deleteButtonIconColor: Int = CometChatTheme.getIconTintSecondary(context)
        @ColorInt private var deleteButtonBackgroundColor: Int = android.graphics.Color.TRANSPARENT
        
        // Send button styling
        @ColorInt private var sendButtonIconColor: Int = CometChatTheme.getColorWhite(context)
        @ColorInt private var sendButtonBackgroundColor: Int = CometChatTheme.getPrimaryColor(context)
        
        // Mic/Resume button styling
        @ColorInt private var micButtonIconColor: Int = CometChatTheme.getIconTintPrimary(context)
        @ColorInt private var micButtonBackgroundColor: Int = android.graphics.Color.TRANSPARENT
        
        // Icon drawables
        private var playIcon: Drawable? = null
        private var pauseIcon: Drawable? = null
        private var deleteIcon: Drawable? = null
        private var sendIcon: Drawable? = null
        private var micIcon: Drawable? = null

        // Container setters
        fun backgroundColor(@ColorInt color: Int) = apply { backgroundColor = color }
        fun cornerRadius(@Dimension radius: Int) = apply { cornerRadius = radius }
        fun strokeColor(@ColorInt color: Int) = apply { strokeColor = color }
        fun strokeWidth(@Dimension width: Int) = apply { strokeWidth = width }
        
        // Waveform setters
        fun barColor(@ColorInt color: Int) = apply { barColor = color }
        fun recordingBarColor(@ColorInt color: Int) = apply { recordingBarColor = color }
        fun playingBarColor(@ColorInt color: Int) = apply { playingBarColor = color }
        fun barWidth(@Dimension width: Int) = apply { barWidth = width }
        fun barSpacing(@Dimension spacing: Int) = apply { barSpacing = spacing }
        fun barMinHeight(@Dimension height: Int) = apply { barMinHeight = height }
        fun barMaxHeight(@Dimension height: Int) = apply { barMaxHeight = height }
        fun barCornerRadius(@Dimension radius: Int) = apply { barCornerRadius = radius }
        fun barCount(count: Int) = apply { barCount = count }
        
        // Duration text setters
        fun durationTextColor(@ColorInt color: Int) = apply { durationTextColor = color }
        fun durationTextAppearance(@StyleRes appearance: Int) = apply { durationTextAppearance = appearance }
        
        // Record button setters
        fun recordButtonIconColor(@ColorInt color: Int) = apply { recordButtonIconColor = color }
        fun recordButtonBackgroundColor(@ColorInt color: Int) = apply { recordButtonBackgroundColor = color }
        fun recordingIndicatorColor(@ColorInt color: Int) = apply { recordingIndicatorColor = color }
        
        // Play button setters
        fun playButtonIconColor(@ColorInt color: Int) = apply { playButtonIconColor = color }
        fun playButtonBackgroundColor(@ColorInt color: Int) = apply { playButtonBackgroundColor = color }
        
        // Pause button setters
        fun pauseButtonIconColor(@ColorInt color: Int) = apply { pauseButtonIconColor = color }
        fun pauseButtonBackgroundColor(@ColorInt color: Int) = apply { pauseButtonBackgroundColor = color }
        
        // Delete button setters
        fun deleteButtonIconColor(@ColorInt color: Int) = apply { deleteButtonIconColor = color }
        fun deleteButtonBackgroundColor(@ColorInt color: Int) = apply { deleteButtonBackgroundColor = color }
        
        // Send button setters
        fun sendButtonIconColor(@ColorInt color: Int) = apply { sendButtonIconColor = color }
        fun sendButtonBackgroundColor(@ColorInt color: Int) = apply { sendButtonBackgroundColor = color }
        
        // Mic button setters
        fun micButtonIconColor(@ColorInt color: Int) = apply { micButtonIconColor = color }
        fun micButtonBackgroundColor(@ColorInt color: Int) = apply { micButtonBackgroundColor = color }
        
        // Icon drawable setters
        fun playIcon(icon: Drawable?) = apply { playIcon = icon }
        fun pauseIcon(icon: Drawable?) = apply { pauseIcon = icon }
        fun deleteIcon(icon: Drawable?) = apply { deleteIcon = icon }
        fun sendIcon(icon: Drawable?) = apply { sendIcon = icon }
        fun micIcon(icon: Drawable?) = apply { micIcon = icon }

        /**
         * Builds the CometChatInlineAudioRecorderStyle instance.
         */
        fun build() = CometChatInlineAudioRecorderStyle(
            backgroundColor = backgroundColor,
            cornerRadius = cornerRadius,
            strokeColor = strokeColor,
            strokeWidth = strokeWidth,
            barColor = barColor,
            recordingBarColor = recordingBarColor,
            playingBarColor = playingBarColor,
            barWidth = barWidth,
            barSpacing = barSpacing,
            barMinHeight = barMinHeight,
            barMaxHeight = barMaxHeight,
            barCornerRadius = barCornerRadius,
            barCount = barCount,
            durationTextColor = durationTextColor,
            durationTextAppearance = durationTextAppearance,
            recordButtonIconColor = recordButtonIconColor,
            recordButtonBackgroundColor = recordButtonBackgroundColor,
            recordingIndicatorColor = recordingIndicatorColor,
            playButtonIconColor = playButtonIconColor,
            playButtonBackgroundColor = playButtonBackgroundColor,
            pauseButtonIconColor = pauseButtonIconColor,
            pauseButtonBackgroundColor = pauseButtonBackgroundColor,
            deleteButtonIconColor = deleteButtonIconColor,
            deleteButtonBackgroundColor = deleteButtonBackgroundColor,
            sendButtonIconColor = sendButtonIconColor,
            sendButtonBackgroundColor = sendButtonBackgroundColor,
            micButtonIconColor = micButtonIconColor,
            micButtonBackgroundColor = micButtonBackgroundColor,
            playIcon = playIcon,
            pauseIcon = pauseIcon,
            deleteIcon = deleteIcon,
            sendIcon = sendIcon,
            micIcon = micIcon
        )
    }

    companion object {
        /**
         * Creates a default style configuration sourcing values from CometChatTheme.
         * All colors and typography are derived from the current theme.
         *
         * **Validates: Requirements 10.9** - Default values from CometChatTheme
         *
         * @param context The Android context for accessing theme attributes
         * @return A fully configured CometChatInlineAudioRecorderStyle with theme defaults
         */
        fun default(context: Context): CometChatInlineAudioRecorderStyle {
            return Builder(context).build()
        }

        /**
         * Creates a new Builder instance for fluent style construction.
         *
         * @param context The Android context for accessing theme attributes
         * @return A new Builder instance with theme defaults
         */
        fun builder(context: Context) = Builder(context)
    }
}
